3.x: NoAnonymousInnerClassesTest fails

When attempting to build the project locally via `./gradlew build`, all of the tests pass except for the `NoAnonymousInnerClassesTest`:
```
io.reactivex.rxjava3.validators.NoAnonymousInnerClassesTest > verify FAILED
    java.lang.AssertionError: Anonymous inner classes found:
    test.io.reactivex.rxjava3.testsupport.SuppressUndeliverableRule$1.class
        at io.reactivex.rxjava3.validators.NoAnonymousInnerClassesTest.verify(NoAnonymousInnerClassesTest.java:125)
```

For context, here are some more details about what I'm running on my machine:
```
$ java --version
openjdk 11.0.2 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
$ gradle --version
------------------------------------------------------------
Gradle 6.5
------------------------------------------------------------

Build time:   2020-06-02 20:46:21 UTC
Revision:     a27f41e4ae5e8a41ab9b19f8dd6d86d7b384dad4

Kotlin:       1.3.72
Groovy:       2.5.11
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          13.0.2 (Oracle Corporation 13.0.2+8)
OS:           Mac OS X 10.15.5 x86_64
```

--- a/src/test/java/io/reactivex/rxjava3/testsupport/SuppressUndeliverableRule.java
+++ b/src/test/java/io/reactivex/rxjava3/testsupport/SuppressUndeliverableRule.java
@@ -30,26 +30,35 @@ import io.reactivex.rxjava3.plugins.RxJavaPlugins;
  */
 public class SuppressUndeliverableRule implements TestRule {
 
+    private static class SuppressUndeliverableRuleStatement extends Statement {
+        private Statement base;
+
+        private SuppressUndeliverableRuleStatement(){}
+        public SuppressUndeliverableRuleStatement(Statement base) {
+            this.base = base;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
+            try {
+                RxJavaPlugins.setErrorHandler(throwable -> {
+                    if (!(throwable instanceof UndeliverableException)) {
+                        throwable.printStackTrace();
+                        Thread currentThread = Thread.currentThread();
+                        currentThread.getUncaughtExceptionHandler().uncaughtException(currentThread, throwable);
+                    }
+                });
+                base.evaluate();
+            } finally {
+                RxJavaPlugins.setErrorHandler(null);
+            }
+        }
+    }
+
     @Override
     public Statement apply(Statement base, Description description) {
-        if (description.getAnnotation(SuppressUndeliverable.class) != null) {
-            return new Statement() {
-                @Override
-                public void evaluate() throws Throwable {
-                    try {
-                        RxJavaPlugins.setErrorHandler(throwable -> {
-                            if (!(throwable instanceof UndeliverableException)) {
-                                throwable.printStackTrace();
-                                Thread currentThread = Thread.currentThread();
-                                currentThread.getUncaughtExceptionHandler().uncaughtException(currentThread, throwable);
-                            }
-                        });
-                        base.evaluate();
-                    } finally {
-                        RxJavaPlugins.setErrorHandler(null);
-                    }
-                }
-            };
+        if (description != null && description.getAnnotation(SuppressUndeliverable.class) != null) {
+            return new SuppressUndeliverableRuleStatement(base);
         } else {
             return base;
         }

